libs/blockscout-service-launcher/src/test/server.rs

```diff
@@ -0,0 +1,76 @@
+use crate::launcher::ServerSettings;
+use rand;
+use reqwest::Url;
+use std::{future::Future, net::SocketAddr, str::FromStr};
+
+pub fn get_test_server_settings() -> (ServerSettings, Url) {
+    let mut server = ServerSettings::default();
+    // Take a random port in range [10000..65535]
+    let port = (rand::random::<u16>() % 55535) + 10000;
+    server.http.addr = SocketAddr::from_str(&format!("127.0.0.1:{port}")).unwrap();
+    server.grpc.enabled = false;
+    let base = Url::parse(&format!("http://{}", server.http.addr)).unwrap();
+    (server, base)
+}
+
+pub async fn init_server<F, R>(run: F, base: &Url, health_check_service: &str)
+where
+    F: FnOnce() -> R + Send + 'static,
+    R: Future<Output = ()> + Send,
+{
+    tokio::spawn(async move { run().await });
+
+    let client = reqwest::Client::new();
+
+    let health_endpoint = base.join("health").unwrap();
+    // Wait for the server to start
+    loop {
+        if let Ok(_response) = client
+            .get(health_endpoint.clone())
+            .query(&[("service", health_check_service)])
+            .send()
+            .await
+        {
+            break;
+        }
+    }
+}
```

Comment:
what if `run().await` failed? this loop wont stop actually, right?
I suggest to use https://docs.rs/reqwest-retry to make requests with total max duration of 10 seconds